docs/14_0_0/components/datatable.md

```diff
@@ -1060,6 +1060,20 @@ JPALazyDataModel<MyEntity> lazyDataModel = JPALazyDataModel.<MyEntity>builder())
         })
         ...
 ```
+### FlowLogix JPALazyDataModel (PrimeFaces Community)
+`JPALazyDataModel` implementation that's fully integrated with Jakarta EE and `@Inject`able. It does not require optional configuration lambdas to be serializable, helping with usability.
```

Comment:
> It does not require optional configuration lambdas to be serializable, helping with usability.

i would just remove that. Otherwise please explain exactly why your lambdas dont need to be serializable.
Even if they are not serialized, not sure how this improves usability.

everything else looks ok for me.

Reply:
Im not sure that explanation in this blurb is quite necessary there. (it's quite lengthy)
Basics about how to deal with final transient variables are described in this article: https://medium.com/@lprimak/how-to-deal-with-non-serializable-fields-in-java-correctly-4ffecad98f15

As you can see here, https://github.com/flowlogix/flowlogix/blob/a915f3343ce9b5e6afac8261cd15d2cbe289d75a/jakarta-ee/flowlogix-datamodel/src/main/java/com/flowlogix/jeedao/primefaces/internal/JPAModelImpl.java#L102

All configuration lambdas are marked as transient and final, thus don't need to be serializable. The only two lambdas that are required to be serializable are builder itself and (optionally-passed) entity manager.
During deserialization builder is invoked and all the config lambdas are rebuilt locally.

This way you get a lot less issues on accidentally bring in non-serializable constituents in the config lambdas.

What do you think?

Reply:
In general i think we should just remove that statement. Everything else makes our implementation bad - but - its just not a problem at all and doesnt improve usability. Our lambdas are serialized, they dont have member fields like inner classes could have. So no real big state or anything will be serialized.

 just would like to understand your statements here because basically its just like this:
- JPALazyDataModel is serializable and if its placed inside a e.g. ViewScoped bean, it will be serialized
- Therefore currently our lambdas are serialized. If they are transient, they would be null after deserialization.
so where do you "store" this to overcame this, if its transient on your side?

Reply:
Did you read the medium article??

Reply:
@lprimak i read the medium article and I don't understand how it solves the problem?

So if you have a transient field NAME and the value is "melloware".  When the EE container decides to serialize it for say clustering Between servers because the value is transient it is now lost?

Your solution seems like it only gets around Serialization errors but doesn't actually solve the serialization problem. Am I not reading that correctly?

Reply:
> Am I not reading that correctly?

:) no... Do you understand what `readResolve()` does? The article (IMHO) pretty clearly explains how it is able to restore the transient final variables based on the current state of the deserialized object, using Lombok's toBuilder().

JPALazyDataModel code does `readResolve()` and then executes the resulting builder to reconstitute al the other config lambdas without having to serialize them.

Reply:
I may write another test at some point to demonstrate this.
This functionality came out from actual applications not working. For example, the config lambdas can refer to `@RequestScoped` objects, or other non-serializable fields and method references inside `@ViewScope`d objects, which are not serializable.

Reply:
I read it and I don't see how it possible solves the problem. If I have two servers and the EE container needs to serialize and send over the meteor so my second server gets the object and deserializes it there is NO way it can have the value "Melloware" and nothing Lombok can do can fix that?

If your solution does work in that scenario then I don't understand serialization.

Reply:
Ok just read this https://www.baeldung.com/java-serialization-readobject-vs-readresolve

It has now convinced me that what you are doing with readResolve is violating the intention of transient and serialization to "work around" the issue of properly designing your objects.  

I would personally NOT recommend this approach and tout it as a benefit it's really a hack.

Reply:
Ok, great, I think you now understand the solution.

> developers and allowing them to be lazy.

Yes!!! The framework needs to do the "heavy lifting" and allow the app developers to be "lazy"

> simply masking potential issues

Can you elaborate on that? What "potential issues" are you talking about exactly?

In any case, this solution wasn't "invented" by me. Quarkus, Weld/CDI and Spring all use it to "fix" constructor injection after deserialization. This is actually a well-known method to fix objects after deserialization and existed well before 2011 when I started using it for this purpose.

Again, I want to reiterate, this does not solve a "hypothetical problem" but has worked for years with actual production applications and countless screens. These lessons have been learned over the last 10 years or so. Well before "native" `JPALazyDataModel` appeared.

Reply:
I have added a test demonstrating how you can use non-serializable class (e.g. RequestScoped) with config lambdas: https://github.com/flowlogix/flowlogix/blob/0b4e3ab2f5aae67cfcb4d63341bb730899742de2/jakarta-ee/flowlogix-datamodel/src/test/java/com/flowlogix/jeedao/primefaces/ModelTest.java#L384

Reply:
I didnt read yet article in detail yet (it's weekend), i just wanted to hear 2-3 sentences where you store/restore your lambdas. no enough concentration on the couch via smartphone to read code examples.

i think we just should remove those sentences from the docs, as it makes our impl more bad. If yours is better, thats ok, you can just not expect that the main project adds this to the docs.
its like: hey users, our impl is not perfect, please use another community project.

if we have a problem here, we can fix it when we get the first issue in the issue tracker. You are also really welcome to contribute here.